lib/cereal/grpc/grpc.go

```diff
@@ -350,22 +347,24 @@ func (g *GrpcBackend) DequeueTask(ctx context.Context, taskName string) (*backen
 	taskCtx, cancel := context.WithCancel(ctx)
 	doneChan := make(chan error, 1)
 	go func() {
-		for {
-			msg, err := s.Recv()
-			if err != nil {
-				logrus.WithError(err).Debug("received error: canceling task context")
-				cancel()
-				doneChan <- err
-				return
-			}
-			if c := msg.GetCancel(); c != nil {
-				logrus.Debug("received cancel: canceling task context")
-				cancel()
-				doneChan <- errors.New("canceled")
-				return
-			}
+		// This goroutine will read the next, and what must be the last
+		// message.
+		var errOut error
+		msg, err := s.Recv()
+		if err != nil {
+			logrus.WithError(err).Debug("received error: canceling task context")
+			errOut = err
+		} else if c := msg.GetCancel(); c != nil {
+			logrus.Debug("received cancel: canceling task context")
+			errOut = context.Canceled
+		} else if c := msg.GetCommitted(); c != nil {
+			logrus.Debug("received comitted")
+			errOut = nil
+		} else {
+			errOut = errUnknownMessage
```

Comment:
[nit] might be nice to used a typed error that allows you to wrap the error message that was received so that that we don't lose context of what actually happened

Reply:
there was no error in this branch. we received a proto message we didn't know how to interpret